### PR TITLE
Sonnet V1.2.7

### PR DESCRIPTION
**Pull request by ultrabear**
Bruh I didn't think dpy2.0 would take this long to come out

### CODE REVIEW

**Comment by ultrabear** [comment]
# Changelog
## Frontend:
- Add(starboard-forceboard): allows forcing a message to be starboarded
- Change(mute): under gotools parser, 5m2s type parsing is allowed now
## Under the Hood:
- Change(search-infractons): Improve infraction renderer to show most amount of chars possible
- Add: gotools library to handle time parsing
- Add: gotools GenerateCacheFile (10x as fast as python variant)
- Change: move alot of code to use io.StringIO
- Add: dpy-stubs typing to codebase
- Change: remove --implicit-reexport mypy flag as its no longer needed
- LeXdPyK 1.4: support for dpy typing, faster ramfs multi dir lookups
- Add(lib_sonnetconfig): Support for dynamic config loading
- Change: reformat wordlist to be alphabetic, add new words
- Change(parsedate): parsedate functions now take optional datetime to conform to dpy typing
- Add: Starting stubs for a new database version, this is not finalized
## Runtime Environment:
- Add: new configs without breaking old config files: `GOLIB_LOAD` and `GOLIB_VERSION`
## Bugs:
- Patch: Remove alot of race conditions and channel type errors by adding dpy typing
- Patch(help): Help command total commands count was not correct
- Patch(cloader): Make cloader differentiate between processing error and FileNotFound
- Patch(update_log_channel): now checks for type of channel, must be TextChannel

File: libs/lib_loaders.py [hunk 1]
```diff
@@ -182,18 +194,25 @@ def load_message_config(guild_id: int, ramfs: lexdpyk.ram_filesystem, datatypes:
...
+                raise FileNotFoundError("No such file")
             else:
                 raise RuntimeError("Wordlist generator received fatal status")
+
         else:
+
             with open("datastore/wordlist.cache.db", "rb") as words:
                 chunksize = words.read(1)[0]
-                num_words = (words.seek(0, io.SEEK_END) - 1) // chunksize
+                num_words = ((words.seek(0, io.SEEK_END) or 0) - 1) // chunksize
```
Comment (Saragon4005) [inline libs/lib_loaders.py:215]: what does this do and why?

File: dlibs/dlib_userupdate.py [hunk 2]
```diff
@@ -95,7 +95,7 @@ async def on_member_join(member: discord.Member, **kargs: Any) -> None:
     # Handle notifer logging
     if member.id in notifier_cache["notifier-log-users"]:
         issues.append("User")
-    if abs(datetime.utcnow().timestamp() - member.created_at.timestamp()) < int(notifier_cache["notifier-log-timestamp"]):
+    if abs(datetime_now().timestamp() - member.created_at.timestamp()) < int(notifier_cache["notifier-log-timestamp"]):
```
Comment (Saragon4005) [inline dlibs/dlib_userupdate.py:98]:
```suggestion
    if abs(datetime.utcnow().timestamp() - member.created_at.timestamp()) < int(notifier_cache["notifier-log-timestamp"]):
```

**Review by Saragon4005 — CHANGES_REQUESTED** [review]

**Review by Saragon4005 — APPROVED** [review]

**Comment by ultrabear** [comment]
Note: there are very few features added this version, because sonnet core is now a stable base, see sonnet-modules for new ideas and tools.

**Comment by ultrabear** [comment]
This PR took so long to approve that discord.py discord.died :sob:

**Pull request merged**